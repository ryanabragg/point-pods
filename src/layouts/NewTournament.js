import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import { withNotification } from '../components/Notification';
import AppMenu from '../components/AppMenu';
import TournamentSettings from '../components/TournamentSettings';

const NotifiedTournamentSettings = withNotification(TournamentSettings);

const styles = theme => ({
  actionButton: {
    margin: 2 * theme.spacing.unit,
    position: 'absolute',
    bottom: 0,
    right: 0,
  },
});

class NewTournament extends Component {
  handleTournamentCreated = (tournament) => {
    this.props.history.push(`/tournament/${tournament.id}`);
  };

  render() {
    const { classes, ...rest } = this.props;
    return (
      <div>
        <AppMenu title='Point Pods' />
        <NotifiedTournamentSettings {...rest}
          title='New Tournament'
          onSubmit={this.handleTournamentCreated}
        />
      </div>
    );
  }
}

NewTournament.defaultProps = {
  history: {
    push: () => null,
    replace: () => null,
  },
  match: {
    params: {
      id: null,
    },
  },
};

NewTournament.propTypes = {
  classes: PropTypes.object.isRequired, // added by withStyles
  history: PropTypes.object, // added by parent
  match: PropTypes.object, // added by parent
};

export default withStyles(styles, { withTheme: true })(NewTournament);
