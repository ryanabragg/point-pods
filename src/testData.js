import api from './api';

const testSettings = {
  pairingMethod: api.pairingMethods.HISTORIC,
  pairingMethodInitial: api.pairingMethods.HISTORIC,
  podSizeMinimum: 4,
  podSizeMaximum: 7,
};

const testPlayers = [{
  id: 'mock0',
  name: 'test0',
  points: 0,
}, {
  id: 'mock1',
  name: 'test1',
  points: 1,
}, {
  id: 'mock2',
  name: 'test2',
  points: 2,
}, {
  id: 'mock3',
  name: 'test3',
  points: 3,
}, {
  id: 'mock4',
  name: 'test4',
  points: 4,
}, {
  id: 'mock5',
  name: 'test5',
  points: 5,
}, {
  id: 'mock6',
  name: 'test6',
  points: 6,
}, {
  id: 'mock7',
  name: 'test7',
  points: 7,
}, {
  id: 'mock8',
  name: 'test8',
  points: 8,
}, {
  id: 'mock9',
  name: 'test9',
  points: 9,
}];

const testTournaments = [{
  id: 'mock0',
  name: 'test0',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Etiam urna tortor, laoreet vitae commodo porttitor, mattis id purus.',
  date: new Date('2018-01-01'),
  done: false,
  staging: true,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 3,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 3 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 0,
    participated: false,
    dropped: false,
  }],
}, {
  id: 'mock1',
  name: 'test1',
  category: 'cat1',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed molestie suscipit suscipit. Nullam sollicitudin tempus pretium.',
  date: new Date('2018-02-01'),
  done: true,
  staging: false,
  pairingMethod: api.pairingMethods.POINTS,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 5,
  rounds: 2,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 0 },
    2: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 1 },
    2: { pod: 2, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
    2: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 3 },
    2: { pod: 2, points: 3 },
  }, {
    id: 'mock4',
    name: 'test4',
    points: 4,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 4 },
    2: { pod: 1, points: 4 },
  }, {
    id: 'mock5',
    name: 'test5',
    points: 5,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 5 },
    2: { pod: 2, points: 5 },
  }, {
    id: 'mock6',
    name: 'test6',
    points: 6,
    participated: true,
    dropped: false,
    2: { pod: 2, points: 6 },
  }],
}, {
  id: 'mock2',
  name: 'test2',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Sed ullamcorper metus vitae nisi rhoncus placerat.',
  date: new Date('2018-01-01'),
  done: false,
  staging: false,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 3 },
  }],
}];

const mockRoute = {
  history: {
    push: jest.fn(to => to),
    replace: jest.fn(to => to),
  },
  match: {
    params: {},
  },
  matchWithInvalid: {
    params: {
      id: 'invalid',
    },
  },
  matchWithID0: {
    params: {
      id: 'mock0',
    },
  },
  matchWithID1: {
    params: {
      id: 'mock1',
    },
  },
};

let mockStore = {};
const mockStorage = {
  setItem: jest.fn((key, val) => Promise.resolve((mockStore[key] = val))),
  getItem: jest.fn(key => Promise.resolve(mockStore[key])),
  removeItem: jest.fn(key => { delete mockStore[key]; Promise.resolve(); }),
  clear: jest.fn(() => Promise.resolve((mockStore = {}))),
  keys: jest.fn(() => Promise.resolve(Object.keys(mockStore))),
  length: jest.fn(() => Promise.resolve(Object.keys(mockStore).length)),
};

let defaultStore = null;
function mockAPI(api, storage) {
  defaultStore = api.store;
  api.store = storage;
  mockStore = {
    settings: testSettings,
    players: testPlayers,
    tournaments: testTournaments,
  };
}
function demockAPI(api) {
  api.store = defaultStore;
  defaultStore = null;
}

export {
  testSettings,
  testPlayers,
  testTournaments,
  mockRoute,
  mockStore,
  mockStorage,
  mockAPI,
  demockAPI,
};
