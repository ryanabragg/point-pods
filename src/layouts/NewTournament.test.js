import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import NewTournament from './NewTournament';

import { mockRoute } from '../testData';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NewTournament />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', () => {
  let component = renderer.create(<NewTournament />);
  expect(component.toJSON()).toMatchSnapshot();
  component = renderer.create(<NewTournament history={mockRoute.history} match={mockRoute.matchWithID1}/>);
  expect(component.toJSON()).toMatchSnapshot();
});
