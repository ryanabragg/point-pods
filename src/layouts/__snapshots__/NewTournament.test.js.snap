// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div>
  <div
    className="AppMenu-root-188"
  >
    <header
      className="MuiPaper-root-203 MuiPaper-elevation4-209 MuiAppBar-root-195 MuiAppBar-positionFixed-196 MuiAppBar-colorPrimary-201 mui-fixed"
    >
      <div
        className="MuiToolbar-root-230 MuiToolbar-regular-232 MuiToolbar-gutters-231 AppMenu-nowrap-190"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-240 MuiIconButton-root-234 MuiIconButton-colorInherit-235 AppMenu-menuButton-191"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-239"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-243"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-250"
          />
        </button>
        <h2
          className="MuiTypography-root-257 MuiTypography-title-263 MuiTypography-colorInherit-276 AppMenu-flex-189"
        >
          Point Pods
        </h2>
      </div>
    </header>
    <div
      className="AppMenu-spacer-193"
    />
  </div>
  <div
    className="NewTournament-center-187"
  >
    <div
      className="MuiPaper-root-203 MuiPaper-elevation1-206 MuiPaper-rounded-204 MuiCard-root-299 APIComponent-root-295"
    >
      <div
        className="MuiCardHeader-root-300"
      >
        <div
          className="MuiCardHeader-content-303"
        >
          <span
            className="MuiTypography-root-257 MuiTypography-headline-262 MuiCardHeader-title-304"
          >
            New Tournament
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root-306"
      />
      <div
        className="MuiCardContent-root-310 APIComponent-content-296"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 MuiFormControl-fullWidth-314 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-name"
          >
             Name
          </label>
          <div
            className="MuiInput-root-325 MuiInput-fullWidth-332 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={true}
              className="MuiInput-input-333 MuiInput-inputMarginDense-334"
              defaultValue={undefined}
              disabled={false}
              id="tournament-name"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 MuiFormControl-fullWidth-314 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-description"
          >
            Description
          </label>
          <div
            className="MuiInput-root-325 MuiInput-fullWidth-332 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-333 MuiInput-inputMarginDense-334"
              defaultValue={undefined}
              disabled={false}
              id="tournament-description"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-category"
          >
            Category
          </label>
          <div
            className="MuiInput-root-325 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <div
              className="MuiSelect-root-338"
            >
              <div
                aria-haspopup="true"
                aria-owns={null}
                aria-pressed="false"
                className="MuiSelect-select-339 MuiSelect-selectMenu-340 MuiInput-input-333 MuiInput-inputMarginDense-334"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id="tournament-category"
                name={undefined}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-243 MuiSelect-icon-342"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-shrink-318 MuiInputLabel-marginDense-317"
            data-shrink={true}
            htmlFor="tournament-date"
          >
            Date
          </label>
          <div
            className="MuiInput-root-325 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-333 MuiInput-inputType-336 MuiInput-inputMarginDense-334"
              defaultValue={undefined}
              disabled={false}
              id="tournament-date"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="date"
              value="2018-07-30"
            />
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-pairingMethod"
          >
            Pairing Method
          </label>
          <div
            className="MuiInput-root-325 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <div
              className="MuiSelect-root-338"
            >
              <div
                aria-haspopup="true"
                aria-owns={null}
                aria-pressed="false"
                className="MuiSelect-select-339 MuiSelect-selectMenu-340 MuiInput-input-333 MuiInput-inputMarginDense-334"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id="tournament-pairingMethod"
                name={undefined}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-243 MuiSelect-icon-342"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-pairingMethodInitial"
          >
            Initial Pairing Method
          </label>
          <div
            className="MuiInput-root-325 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <div
              className="MuiSelect-root-338"
            >
              <div
                aria-haspopup="true"
                aria-owns={null}
                aria-pressed="false"
                className="MuiSelect-select-339 MuiSelect-selectMenu-340 MuiInput-input-333 MuiInput-inputMarginDense-334"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id="tournament-pairingMethodInitial"
                name={undefined}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-243 MuiSelect-icon-342"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-podSizeMinimum"
          >
            Minimum Pod Size
          </label>
          <div
            className="MuiInput-root-325 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-333 MuiInput-inputType-336 MuiInput-inputMarginDense-334"
              defaultValue={undefined}
              disabled={false}
              id="tournament-podSizeMinimum"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 APIComponent-field-297"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-podSizeMaximum"
          >
            Maximum Pod Size
          </label>
          <div
            className="MuiInput-root-325 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-333 MuiInput-inputType-336 MuiInput-inputMarginDense-334"
              defaultValue={undefined}
              disabled={false}
              id="tournament-podSizeMaximum"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <hr
        className="MuiDivider-root-306"
      />
      <div
        className="MuiCardActions-root-345"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-311 MuiFormControl-marginDense-313 MuiCardActions-action-346"
        >
          <label
            className="MuiFormLabel-root-320 MuiInputLabel-root-315 MuiInputLabel-formControl-316 MuiInputLabel-animated-319 MuiInputLabel-marginDense-317"
            data-shrink={false}
            htmlFor="tournament-newCategory"
          >
            Add Category
          </label>
          <div
            className="MuiInput-root-325 MuiInput-formControl-326 MuiInput-underline-329"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-333 MuiInput-inputMarginDense-334"
              defaultValue={undefined}
              disabled={false}
              id="tournament-newCategory"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          aria-label="Add Category"
          className="MuiButtonBase-root-240 MuiIconButton-root-234 MuiCardActions-action-346"
          disabled={false}
          id="tournament-newCategory-add"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-239"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-243"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-250"
          />
        </button>
        <button
          aria-label="Create"
          className="MuiButtonBase-root-240 MuiButton-root-347 MuiButton-text-349 MuiButton-flat-352 APIComponent-right-298 MuiCardActions-action-346"
          disabled={false}
          id="tournament-submit"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-348"
          >
            Create
          </span>
          <span
            className="MuiTouchRipple-root-250"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 2`] = `
<div>
  <div
    className="AppMenu-root-188"
  >
    <header
      className="MuiPaper-root-203 MuiPaper-elevation4-209 MuiAppBar-root-195 MuiAppBar-positionFixed-196 MuiAppBar-colorPrimary-201 mui-fixed"
    >
      <div
        className="MuiToolbar-root-230 MuiToolbar-regular-232 MuiToolbar-gutters-231 AppMenu-nowrap-190"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-240 MuiIconButton-root-234 MuiIconButton-colorInherit-235 AppMenu-menuButton-191"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-239"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-243"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-250"
          />
        </button>
        <h2
          className="MuiTypography-root-257 MuiTypography-title-263 MuiTypography-colorInherit-276 AppMenu-flex-189"
        >
          Point Pods
        </h2>
      </div>
    </header>
    <div
      className="AppMenu-spacer-193"
    />
  </div>
  <a
    aria-label="Start Tournament"
    className="MuiButtonBase-root-240 MuiButton-root-347 MuiButton-fab-362 MuiButton-contained-356 MuiButton-containedSecondary-358 MuiButton-raised-359 MuiButton-raisedSecondary-361 NewTournament-actionButton-186"
    href="/tournament/mock1"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-348"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-243"
        color={undefined}
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-250"
    />
  </a>
  <div>
    <div
      className="MuiToolbar-root-230 MuiToolbar-regular-232 MuiToolbar-gutters-231"
    >
      <h1
        className="MuiTypography-root-257 MuiTypography-headline-262"
      >
        Add Players
      </h1>
      <p
        className="MuiTypography-root-257 MuiTypography-body1-266"
      />
      <button
        aria-haspopup="true"
        aria-label="Players Menu"
        aria-owns={null}
        className="MuiButtonBase-root-240 MuiButton-root-347 MuiButton-text-349 MuiButton-flat-352 APIComponent-right-371"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-348"
        >
          0 Players
        </span>
        <span
          className="MuiTouchRipple-root-250"
        />
      </button>
    </div>
    <div
      className="APIComponent-addPlayer-372"
    >
      <div
        className="IntegrationAutosuggest-root-374"
      >
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-1"
          className="IntegrationAutosuggest-container-375"
          role="combobox"
        >
          <div
            aria-describedby={undefined}
            className="MuiFormControl-root-311 MuiFormControl-fullWidth-314"
          >
            <div
              aria-activedescendant={null}
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              className="MuiInput-root-325 MuiInput-fullWidth-332 MuiInput-formControl-326 MuiInput-underline-329"
              style={Object {}}
            >
              <input
                aria-invalid={false}
                autoComplete="off"
                autoFocus={undefined}
                className="MuiInput-input-333"
                defaultValue={undefined}
                disabled={false}
                id={undefined}
                name={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={undefined}
                placeholder="Add a player"
                readOnly={undefined}
                required={false}
                rows={undefined}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="MuiPaper-root-203 MuiPaper-elevation2-207"
            id="react-autowhatever-1"
            role="listbox"
            style={Object {}}
          />
        </div>
        <button
          className="MuiButtonBase-root-240 MuiIconButton-root-234 MuiIconButton-colorInherit-235"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-239"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-243"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-250"
          />
        </button>
      </div>
    </div>
    <div
      className="APIComponent-list-373"
    >
      <ul
        className="MuiList-root-379 MuiList-padding-380"
      />
    </div>
  </div>
</div>
`;
